src/statistics/statistics.c

```diff
@@ -533,114 +511,175 @@ void statistics_stop(int exit_code) {
 			}
 
 			fflush(f);
-		}
-
-		// TODO: quando reintegriamo la parte distribuita, qui si deve fare la riduzione tra tutti i kernel
 
-		// Write the final outcome of the simulation on screen
-		if (master_kernel() && master_thread()) {
-			if(exit_code == EXIT_FAILURE) {
-				printf("\n--------- SIMULATION ABNORMALLY TERMINATED ----------\n");
-			}
-			if(exit_code == EXIT_SUCCESS) {
-				printf("\n--------- SIMULATION CORRECTLY COMPLETED ----------\n");
+			if(master_kernel()){
+				// TODO: do the statistics reduction between all the MPI kernels in another file
```

Comment:
Any chance to have this implemented? :smile: 
Basically this entails having an `MPI_Reduce` in rank 0 (before checking if I am the master kernel) to compute the average of `system_wide_stats` coming from all ranks.
Given that we're dealing with a struct and we want to compute the average, we should define beforehand a new reduction operation using `MPI_Op_create`, maybe when initializing the statistics subsystem.